src/core/StepRecorder.cpp

```diff
@@ -288,26 +288,26 @@ void StepRecorder::removeNotesReleasedForTooLong()
 	int nextTimout = std::numeric_limits<int>::max();
 	bool notesRemoved = false;
 
-	QMutableVectorIterator<StepNote*> itr(m_curStepNotes);
-	while (itr.hasNext())
+	m_curStepNotes.erase(std::remove_if(m_curStepNotes.begin(), m_curStepNotes.end(), [&](StepNote* stepNote) 
 	{
-		StepNote* stepNote = itr.next();
-
-		if(stepNote->isReleased())
+		bool shouldRemove = false;
+		if (stepNote->isReleased()) 
 		{
-			const int timeSinceReleased = stepNote->timeSinceReleased(); // capture value to avoid wraparound when calculting nextTimout
-			if (timeSinceReleased >= REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS)
+			const int timeSinceReleased = stepNote->timeSinceReleased();
+			shouldRemove = timeSinceReleased >= REMOVE_RELEASED_NOTE_TIME_THRESHOLD_MS;
+
+			if (shouldRemove) 
 			{
 				delete stepNote;
-				itr.remove();
 				notesRemoved = true;
```

Comment:
We may remove this and check the return value of `std::remove_if` instead, if we stick to `std::remove_if`.